### DISCUSSION TITLE
Refresh Token

### DISCUSSION THREAD

**Discussion by jaredcosulich** [supabase/supabase#7637] [Questions]
I have a relatively complex situation where I want to log a user in via a hidden iframe. The user comes in via a magic link, is redirected to another website where a hidden iframe is loaded which provides authentication information to the third-party website. 

On most browsers once the user is authenticated on the main site they will also be authenticated when in the iframe. On Brave though it seems this is not the case. I'd like to be able to pass information to the third-party site that can be passed into the iframe to authenticate the user.

If I pass the session's refresh_token to the third party site can that be used to authenticate the user in the iframe? Will that log them out of the primary website? Is there a better way for the primary website to pass information to the third-party website that will allow the third-party website to authenticate the user?

**Comment by GaryAustin1** [comment]
I'm not commenting on your process and iframes.   But if you use a refresh token in a 2nd environment, that refresh token is no longer valid in any other context.  The jwt's live on until they expire, but can't be refreshed unless you have a refresh token chain used only once from sign in.

**Reply by jaredcosulich** [reply]
Can you describe what a "refresh token chain" means exactly or is there somewhere to read more about that?

Is there a better way to pass information to another context that would allow us to authenticate that user in the second context automatically?

**Reply by GaryAustin1** [reply]
There may be some info on it, but I'm out for the night and don't have a link for you.
Basically when you sign in you get a jwt token good for 1 hour (by default) and a refresh token, both in the session.  Every time you use that refresh token to get a new jwt, it gets invalidated and you get a new one with the new jwt.  If your user signs in on a 2nd device they get a different refresh token and as they are used for refreshing, they get invalidated.
So you can have multiple sign in "chains" of refresh tokens.
If you sign out, ALL current tokens are invalidated so a user has to log in on any device they may have been on.

**Reply by jaredcosulich** [reply]
Ok, great, this is helpful. Thank you.
